src/main/kotlin/no/nav/familie/ef/mottak/service/MappeService.kt

```diff
@@ -0,0 +1,111 @@
+            return null
+        }
+        val mapper = finnMapperForEnhet(enhetsnummer)
+        val søkestreng = finnSøkestrengForSøknad(søknadId)
+        return utledMappeForSøkestreng(mapper, søkestreng)?.id?.toLong()
+    }
+
+    private fun finnMapperForEnhet(enhetsnummer: String): FinnMappeResponseDto {
+        val finnMappeRequest = FinnMappeRequest(
+            tema = listOf(),
+            enhetsnr = enhetsnummer,
+            opprettetFom = null,
+            limit = 1000
+        )
+        val mapperResponse = integrasjonerClient.finnMappe(finnMappeRequest)
+        log.info("Mapper funnet: Antall: ${mapperResponse.antallTreffTotalt}, ${mapperResponse.mapper} ")
+
+        return mapperResponse
+    }
+
+    private fun finnSøkestrengForSøknad(søknadId: String): MappeSøkestreng {
+        val søknad = søknadService.get(søknadId)
+
+        return when (dokumenttypeTilStønadType(søknad.dokumenttype)) {
+            StønadType.OVERGANGSSTØNAD -> mappeFraOvergangsstønad(søknad)
+            StønadType.BARNETILSYN -> mappeFraBarnetilsyn(søknad)
+            StønadType.SKOLEPENGER -> OPPLÆRING
+            else -> UPLASSERT
+        }
+    }
+
+    private fun utledMappeForSøkestreng(mapperResponse: FinnMappeResponseDto, søkestreng: MappeSøkestreng): MappeDto? {
+        return if (UPLASSERT == søkestreng) {
+            null
+        } else {
+            mapperResponse.mapper.filter {
+                !it.navn.contains("EF Sak", true) &&
+                    it.navn.contains(søkestreng.verdi, true)
+            }.maxByOrNull { it.id }
+                ?: error("Fant ikke mappe for $søkestreng")
+        }
+    }
+
+    private fun mappeFraOvergangsstønad(søknad: Søknad): MappeSøkestreng {
+        val søknadsdata = objectMapper.readValue<SøknadOvergangsstønad>(søknad.søknadJson.data)
+        return when {
+            erSærligTilsynskrevende(søknadsdata) -> SÆRLIG_TILSYNSKREVENDE
+            erSelvstendig(søknadsdata.aktivitet) -> SELVSTENDIG
+            else -> UPLASSERT
+        }
+    }
+
+    private fun erSærligTilsynskrevende(søknadsdata: SøknadOvergangsstønad) =
+        søknadsdata.situasjon.verdi.barnMedSærligeBehov?.verdi != null
+
+    private fun mappeFraBarnetilsyn(søknad: Søknad): MappeSøkestreng {
+        val søknadsdata = objectMapper.readValue<SøknadBarnetilsyn>(søknad.søknadJson.data)
+        return when {
+            erSærligTilsynskrevende(søknadsdata) -> SÆRLIG_TILSYNSKREVENDE
+            erSelvstendig(søknadsdata.aktivitet) -> SELVSTENDIG
+            else -> UPLASSERT
+        }
+    }
+
+    private fun erSærligTilsynskrevende(søknadsdata: SøknadBarnetilsyn) =
+        søknadsdata.barn.verdi.any { it.barnepass?.verdi?.årsakBarnepass?.svarId == "trengerMerPassEnnJevnaldrede" }
+
+    private fun erSelvstendig(aktivitet: Søknadsfelt<Aktivitet>) =
+        aktivitet.verdi.firmaer?.verdi?.isNotEmpty() ?: false ||
+            aktivitet.verdi.virksomhet?.verdi != null
```

Comment:
Er det noen spesiell grunn til at den ene sjekkes for notEmpty, mens den andre sjekkes for not null?

Reply:
Vet ikke, jeg kopierte egentlig bare det som lå i oppgaveservice for at dette skulle bli gjenbrukbart :)